### PR TITLE
skywalking extension create new trace context if sw8 header invalid

### PR DESCRIPTION
**Pull request by StarryVae**
Commit Message: skywalking extension create new trace context if sw8 header invalid
Additional Description: N/A
Risk Level: low
Testing: N/A
Docs Changes: N/A
Release Notes: N/A

### CODE REVIEW

**Comment by StarryVae** [comment]
issue #25842

**Comment by wbpcode** [comment]
Please check the ci first. Thanks. /wait

**Comment by StarryVae** [comment]
> Please check the ci first. Thanks. /wait

done, PTAL, thanks.

**Comment by StarryVae** [comment]
@wbpcode PTAL, thanks.

File: source/extensions/tracers/skywalking/skywalking_tracer_impl.cc [hunk 1]
```diff
@@ -72,8 +72,11 @@ Tracing::SpanPtr Driver::startSpan(const Tracing::Config&, Tracing::TraceContext
           createSpanContext(toStdStringView(header_value_string)); // NOLINT(std::string_view)
       tracing_context = tracing_context_factory_->create(span_context);
     } catch (std::exception& e) {
-      ENVOY_LOG(warn, "New SkyWalking Span/Segment cannot be created for error: {}", e.what());
-      return std::make_unique<Tracing::NullSpan>();
+      ENVOY_LOG(
+          warn,
+          "New SkyWalking Span/Segment with previous span context cannot be created for error: {}",
+          e.what());
+      tracing_context = tracing_context_factory_->create();
```
Comment (wbpcode) [inline source/extensions/tracers/skywalking/skywalking_tracer_impl.cc:79]: If there is no valid propagation header, I think we should downgrade to the case where propagation header is no exist. In that case, the `decision` will be used to decide to create null span or valid span.

**Review by wbpcode — COMMENTED** [review]
Thanks. A commend is added.

File: source/extensions/tracers/skywalking/skywalking_tracer_impl.cc (hunk 1 above)
Comment (StarryVae) [reply source/extensions/tracers/skywalking/skywalking_tracer_impl.cc]: done,thanks,

**Comment by StarryVae** [comment]
@wbpcode PTAL again, thanks.

File: test/extensions/tracers/skywalking/skywalking_tracer_impl_test.cc [hunk 2]
```diff
@@ -136,9 +136,53 @@ TEST_F(SkyWalkingDriverTest, SkyWalkingDriverStartSpanTestWithClientConfig) {
...
+    EXPECT_FALSE(span->tracingContext()->skipAnalysis());
+
+    EXPECT_CALL(*mock_stream_ptr_, sendMessageRaw_(_, _));
+    span->finishSpan();
+
+    EXPECT_EQ(4U, factory_context.store_.counter("tracing.skywalking.segments_sent").value());
+  }
+
+  {
+    // Create root segment span with disabled tracing.
```
Comment (wbpcode) [inline test/extensions/tracers/skywalking/skywalking_tracer_impl_test.cc:180]: Create null span with disabled tracing.

File: test/extensions/tracers/skywalking/skywalking_tracer_impl_test.cc [hunk 3]
```diff
@@ -147,12 +191,13 @@ TEST_F(SkyWalkingDriverTest, SkyWalkingDriverStartSpanTestWithClientConfig) {
   }
 
   {
-    // Create new span segment with error propagation header.
+    // Create new span segment with error propagation header and disabled tracing.
```
Comment (wbpcode) [inline test/extensions/tracers/skywalking/skywalking_tracer_impl_test.cc:194]: ditto.

File: test/extensions/tracers/skywalking/skywalking_tracer_impl_test.cc [hunk 4]
```diff
@@ -164,12 +209,16 @@ TEST_F(SkyWalkingDriverTest, SkyWalkingDriverStartSpanTestWithClientConfig) {
   }
 
   {
-    // Create root segment span with disabled tracing.
+    // Create new span segment with error propagation header and disabled tracing.
```
Comment (wbpcode) [inline test/extensions/tracers/skywalking/skywalking_tracer_impl_test.cc:212]: ditto.

**Review by wbpcode — COMMENTED** [review]
LGTM overall. Some nit comments are added. And I think this change need one line change log to  record the minor behavior change.

File: test/extensions/tracers/skywalking/skywalking_tracer_impl_test.cc (hunk 2 above)
Comment (StarryVae) [reply test/extensions/tracers/skywalking/skywalking_tracer_impl_test.cc]: done

File: test/extensions/tracers/skywalking/skywalking_tracer_impl_test.cc (hunk 3 above)
Comment (StarryVae) [reply test/extensions/tracers/skywalking/skywalking_tracer_impl_test.cc]: done

File: test/extensions/tracers/skywalking/skywalking_tracer_impl_test.cc (hunk 4 above)
Comment (StarryVae) [reply test/extensions/tracers/skywalking/skywalking_tracer_impl_test.cc]: done

**Review by wbpcode — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**